### DISCUSSION TITLE
[BUG]: [Document] Development and debug environment setup

### DISCUSSION THREAD

**Discussion by SARAVANA1501** [dotnet/spark#733] [Q&A]
I just started contributing to dotnet spark repo, I am unable to run complete end to end test on my local, so have to push my code to repo to see the result, like blind folded. 
I just used [this](https://github.com/dotnet/spark/blob/master/docs/developer-guide.md) document for local setup, But lot of version conflict issues, Can we improve this document? or even some simple docker image for local setup will help.

**Comment by indy-3rdman** [comment]
> I just used [this](https://github.com/dotnet/spark/blob/master/docs/developer-guide.md) document for local setup, But lot of version conflict issues, Can we improve this document? or even some simple docker image for local setup will help.

@SARAVANA1501, issue #710 has been opened to provide official docker images. In the meantime it would be great if you could try using my unofficial dev image and let me know if it contains everything you need.
A brief walk-through of how to build .NET for Apache Spark using the image is available [here](https://3rdman.de/2020/10/build-net-for-apache-spark-with-vs-code-in-a-browser/)

**Comment by suhsteve** [comment]
@SARAVANA1501 have you been able to get the [Get Started](https://docs.microsoft.com/en-us/dotnet/spark/tutorials/get-started) example to work ?  If so, then your environment should be set up to be able to run unit tests.

**Comment by rapoth** [comment]
@SARAVANA1501 In addition to what @suhsteve is asking, could you also include the error you are hitting into when you are running the tests?

Finally, do you have any suggestions on how we should improve the docs? (Also, please feel free to open a PR directly if you have any ideas).

To unblock yourself, the docker images from @indy-3rdman would be a great start as well!
